bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -839,7 +839,7 @@ impl<'a> CircuitInputStateRef<'a> {
         let call = self.call()?;
 
         // Return from a call with a failure
-        if step.depth != next_depth && next_result.is_zero() {
+        if step.depth == next_depth + 1 && next_result.is_zero() {
```

Comment:
Would you like to also change line 904 to `step.depth == next_depth + 1`? It seems more strict rules first can help detect such kinds of issue.

Reply:
Good to try updating , thanks

Reply:
updated in [bda71b7](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/630/commits/bda71b728670f0567cf5e8961af6590f7d6b372f)